weblate/glossary/tasks.py

```diff
@@ -40,6 +48,29 @@ def sync_glossary_languages(pk: int, component: Component | None = None) -> None
         component.create_translations_task()
 
 
+@app.task(trail=False, autoretry_for=(Project.DoesNotExist, WeblateLockTimeoutError))
+def cleanup_stale_glossaries(project: int | Project) -> None:
+    if isinstance(project, int):
+        project = Project.objects.get(pk=project)
+
+    languages_in_non_glossary_components: set[int] = set(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=False
+        ).values_list("language_id", flat=True)
+    )
+
+    glossary_translations = prefetch_stats(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=True
+        ).exclude(language__id__in=languages_in_non_glossary_components)
+    )
```

Comment:
I think the removal should be completely avoided in some corner cases:

* There is only a glossary component in the project (it might be shared with other projects or misconfigured)
* Probably it should skip removal for glossary components with a remote repository configured (`repo != "local:"`) because we should avoid removing glossaries that are managed outside Weblate.